collector/thermal_linux.go

```diff
@@ -0,0 +1,72 @@
+// Copyright 2015 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build !nonetstat
+
+package collector
+
+import (
+	"fmt"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/procfs/sysfs"
+)
+
+const (
+	thermalZoneSubsystem = "thermal"
+)
+
+type thermalCollector struct {
+	fs   sysfs.FS
+	temp *prometheus.Desc
+}
+
+func init() {
+	registerCollector("thermal", defaultEnabled, NewThermalCollector)
+}
+
+// NewThermalCollector takes and returns
+// a new Collector exposing thermal stats.
+func NewThermalCollector() (Collector, error) {
+
+	fs, err := sysfs.NewFS(*sysPath)
+	if err != nil {
+		return nil, fmt.Errorf("failed to open sysfs: %v", err)
+	}
+
+	return &thermalCollector{
+		fs: fs,
+		temp: prometheus.NewDesc(
+			prometheus.BuildFQName(namespace, thermalZoneSubsystem, "zones"),
```

Comment:
Instead of "zones", this should probably be "zone_temp_celcius" to clarify what's being measured.  So the resulting metric would look like:
```
node_thermal_zone_temp_celcius{zone="0"} 55
```

Reply:
ok!